ca/certificate-authority_test.go

```diff
@@ -463,6 +491,40 @@ func TestWrongSignature(t *testing.T) {
 	}
 }
 
+func TestProfileSelection(t *testing.T) {
+	ctx := setup(t)
+	defer ctx.cleanUp()
+	ctx.caConfig.MaxNames = 3
+	ca, _ := NewCertificateAuthorityImpl(ctx.caConfig, ctx.fc, ctx.stats, caCert, caKey, ctx.keyPolicy)
+	ca.Publisher = &mocks.Publisher{}
+	ca.PA = ctx.pa
+	ca.SA = ctx.sa
+
+	correctFlags := []x509.KeyUsage{
+		x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		x509.KeyUsageDigitalSignature,
+	}
+	csrs := [][]byte{CNandSANCSR, ECDSACSR}
+	for i, csrDER := range csrs {
+		csr, err := x509.ParseCertificateRequest(csrDER)
+		test.AssertNotError(t, err, "Cannot parse CSR")
+
+		// Sign CSR
+		issuedCert, err := ca.IssueCertificate(*csr, ctx.reg.ID)
+		test.AssertNotError(t, err, "Failed to sign certificate")
+		if err != nil {
```

Comment:
This is unnecessary. `test.AssertNotError` will fail the test, exiting immediately.